File: clojure/raindrops/src/raindrops.clj
 (ns raindrops)

 (def elements {
                3 "Pling"
                5 "Plang"
                7 "Plong"
                })

 (defn convert [n]
   (let [result (apply str (map #(if (zero? (mod n (first %))) (second %) "") elements))]
     (if (clojure.string/blank? result)
       (str n)
       result)))

File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

factors :: [(Int, String)]
factors = [(3, "Pling"),(5, "Plang"),(7, "Plong")]

convertForFactor :: Int -> (Int, String) -> String
convertForFactor n (f, s) = if n `mod` f == 0 then s else ""

convert :: Int -> String
convert n = if v == "" then show n else v
  where v = concatMap (convertForFactor n) factors
